### PR TITLE
(#1016) move nonNulls from input,output,bytes to own packages

### PR DESCRIPTION
**Pull request by mazdack**
all done

### CODE REVIEW

**Comment by 0crat** [comment]
Job [#1067](https://github.com/yegor256/cactoos/issues/1067) is now in scope, role is `REV`

**Comment by 0crat** [comment]
This pull request [#1067](https://github.com/yegor256/cactoos/issues/1067) is assigned to @fabriciofx[/z](https://www.0crat.com/u/fabriciofx), here is [why](https://www.0crat.com/footprint/C63314D6Z/8d750f45-7812-445d-bade-3377b031d1dd); the budget is 15 minutes, see [§4](http://www.zerocracy.com/policy.html#4); please, read [§27](http://www.zerocracy.com/policy.html#27) and when you decide to accept the changes, inform @llorllale[/z](https://www.0crat.com/u/llorllale) (the architect) right in this ticket; if you decide that this PR should not be accepted ever, also inform the architect; this [blog post](http://www.yegor256.com/2015/02/09/serious-code-reviewer.html) will help you understand what is expected from a code reviewer; there will be no monetary reward for this job

**Comment by llorllale** [comment]
@0crat refuse

**Comment by 0crat** [comment]
> @0crat refuse ([here](https://github.com/yegor256/cactoos/issues/1067#issuecomment-468921468))

@llorllale The user @fabriciofx[/z](https://www.0crat.com/u/fabriciofx) resigned from [#1067](https://github.com/yegor256/cactoos/issues/1067), please stop working. Reason for job resignation: Order was cancelled

**Comment by 0crat** [comment]
This pull request [#1067](https://github.com/yegor256/cactoos/issues/1067) is assigned to @iprogrammerr[/z](https://www.0crat.com/u/iprogrammerr), here is [why](https://www.0crat.com/footprint/C63314D6Z/eda62b0c-b0ea-47b3-830b-0ca82be4fa6f); the budget is 15 minutes, see [§4](http://www.zerocracy.com/policy.html#4); please, read [§27](http://www.zerocracy.com/policy.html#27) and when you decide to accept the changes, inform @llorllale[/z](https://www.0crat.com/u/llorllale) (the architect) right in this ticket; if you decide that this PR should not be accepted ever, also inform the architect; this [blog post](http://www.yegor256.com/2015/02/09/serious-code-reviewer.html) will help you understand what is expected from a code reviewer; there will be no monetary reward for this job

**Comment by Iprogrammerr** [comment]
@mazdack Could you remove typo from PR title? As we use cactoos-matchers now, how about replacing
```java
@Test(expected = Exception.class) 
``` 
with Throws from that library?

**Comment by 0crat** [comment]
This pull request [#1067](https://github.com/yegor256/cactoos/issues/1067) is assigned to @victornoel[/z](https://www.0crat.com/u/victornoel), here is [why](https://www.0crat.com/footprint/C63314D6Z/558116c2-56dd-4eb3-b3cc-a530e49df98d); the budget is 15 minutes, see [§4](http://www.zerocracy.com/policy.html#4); please, read [§27](http://www.zerocracy.com/policy.html#27) and when you decide to accept the changes, inform @llorllale[/z](https://www.0crat.com/u/llorllale) (the architect) right in this ticket; if you decide that this PR should not be accepted ever, also inform the architect; this [blog post](http://www.yegor256.com/2015/02/09/serious-code-reviewer.html) will help you understand what is expected from a code reviewer; there will be no monetary reward for this job

**Comment by victornoel** [comment]
@llorllale it's good to merge for me :)

**Comment by llorllale** [comment]
@mazdack @victornoel two new packages were introduced in this PR (`org.cactoos.input`, `org.cactoos.output`); we have not split `org.cactoos.io` yet, see #999

**Comment by victornoel** [comment]
@llorllale indeed, so those this mean that we have to name them `InputNoNulls` and `OutputNotNulls` in `org.cactoos.io` and wait for #999 to move them in their own package?

**Comment by llorllale** [comment]
@victornoel for now, yes, that is correct

**Comment by mazdack** [comment]
@victornoel removed new packages, moved to io
@victornoel fixed typo in PR title. regarding throws, i didn't quite get you. in the docs we find [3 ways to work with expected exceptions](https://github.com/junit-team/junit4/wiki/Exception-testing). i think that current one is cleanest as we don't need to check exception message, just exception class.

File: src/test/java/org/cactoos/InputTest.java [hunk 1]
```diff
@@ -24,27 +24,28 @@
...
+import org.cactoos.io.InputNoNulls;
 import org.junit.Test;
 
 /**
- * Test case for {@link Input.NoNulls}.
+ * Test case for {@link InputNoNulls}.
  * @since 0.10
  * @checkstyle JavadocMethodCheck (500 lines)
  */
 public final class InputTest {
```
Comment (victornoel) [inline src/test/java/org/cactoos/InputTest.java]: @mazdack please rename this to `InputNoNullsTest`

File: src/test/java/org/cactoos/OutputTest.java [hunk 2]
```diff
@@ -24,27 +24,28 @@
...
+import org.cactoos.io.OutputNoNulls;
 import org.junit.Test;
 
 /**
- * Test case for {@link Output.NoNulls}.
+ * Test case for {@link OutputNoNulls}.
  * @since 0.10
  * @checkstyle JavadocMethodCheck (500 lines)
  */
 public final class OutputTest {
```
Comment (victornoel) [inline src/test/java/org/cactoos/OutputTest.java]: @mazdack please rename this to `OutputNoNullsTest`

File: src/test/java/org/cactoos/InputTest.java [hunk 3]
```diff
@@ -24,27 +24,28 @@
 package org.cactoos;
```
Comment (victornoel) [inline src/test/java/org/cactoos/InputTest.java]: @mazdack please move this to `io` package.

File: src/test/java/org/cactoos/OutputTest.java [hunk 4]
```diff
@@ -24,27 +24,28 @@
 package org.cactoos;
```
Comment (victornoel) [inline src/test/java/org/cactoos/OutputTest.java]: @mazdack please move this to `io` package.

**Comment by victornoel** [comment]
@mazdack I didn't say anything about exceptions, it was @Iprogrammerr. I think it was about using https://github.com/llorllale/cactoos-matchers/blob/master/src/main/java/org/llorllale/cactoos/matchers/Throws.java to assert about thrown exceptions instead of using junit annotation. I personally don't think it is needed now, this PR has been in the pipe for long enough, but maybe @llorllale has another point of view?

@mazdack I've added some more little comments for the tests (name and location).

**Comment by mazdack** [comment]
@victornoel pushed fixes to your review.
regarding Throws seems like another task indeed

**Comment by victornoel** [comment]
@llorllale I think we are good now :)
@mazdack thx

**Comment by llorllale** [comment]
@rultor merge

**Comment by 0crat** [comment]
@sereshqua[/z](https://www.0crat.com/u/sereshqua) please review this job completed by @victornoel[/z](https://www.0crat.com/u/victornoel), as in [§30](http://www.zerocracy.com/policy.html#30); the job will be fully closed and all payments will be made when the quality review is completed

**Comment by 0crat** [comment]
Job [#1067](https://github.com/yegor256/cactoos/issues/1067) is not in the agenda of @victornoel[/z](https://www.0crat.com/u/victornoel), can't inspect

**Comment by 0crat** [comment]
Code review was too long (14 days), architects (@llorllale) were penalized, see [§55](http://www.zerocracy.com/policy.html#55)

**Comment by 0crat** [comment]
The job [#1067](https://github.com/yegor256/cactoos/issues/1067) is now out of scope

**Comment by 0crat** [comment]
Payment to `ARC` for a closed pull request, as in [§28](http://www.zerocracy.com/policy.html#28): +10 point(s) just awarded to @llorllale[/z](https://www.0crat.com/u/llorllale)

**Comment by sereshqua** [comment]
@0crat quality acceptable

**Comment by 0crat** [comment]
Quality review completed: +4 point(s) just awarded to @sereshqua[/z](https://www.0crat.com/u/sereshqua)

**Comment by 0crat** [comment]
Order was finished, quality is "acceptable": +15 point(s) just awarded to @victornoel[/z](https://www.0crat.com/u/victornoel)

**Comment by victornoel** [comment]
@sereshqua why only acceptable? Can you explain what should I have done to get good? Thanks :)

**Comment by sereshqua** [comment]
@victornoel 

1) my concern was that @mazdack did not mentioned a person, to whom he was referring her message in PR ([see](https://github.com/yegor256/cactoos/pull/1067#issue-251659222)

2) however, I have checked other PRs and now it seem clear to that in all PRs, NOT tasks, in first messages PR submitter do not mention anyone.

3) so after reviewing it again, I think the quality is good here; so if I could somehow change my verdict, I would put `good` in here.

Sorry for confusion

**Comment by victornoel** [comment]
@sereshqua no problem, I believe it's better to always explicit what is problematic so that we can improve :) also, btw, @mazdack work is evaluated by QA in the issue he solved, not here (even though I think it's a good thing to explicit it here anyway, but it's just my opinion). Thanks for taking the time to explain :)

**Comment by mazdack** [comment]
Didn't know, that I should mention anyone in pr message. Still, the review was very good, helped to make pr much better

2 апр. 2019 г. 10:03 пользователь Serhii Boiko <notifications@github.com> написал:

@victornoel<https://github.com/victornoel>

  1.  my concern was that @mazdack<https://github.com/mazdack> did not mentioned a person, to whom he was referring her message in PR (see<https://github.com/yegor256/cactoos/pull/1067#issue-251659222>

  2.  however, I have checked other PRs and now it seem clear to that in all PRs, NOT tasks, in first messages PR submitter do not mention anyone.

  3.  so after reviewing it again, I think the quality is good here; so if I could somehow change my verdict, I would put good in here.

Sorry for confusion

-

**Comment by sereshqua** [comment]
@victornoel sure, completely agree with, will be doing it!

@mazdack - it is not about technical review, it is about procedures and process review

**Pull request merged**